diff --git a/Makefile b/Makefile
index 9566b48..a6425e9 100644
--- a/Makefile
+++ b/Makefile
@@ -1716,7 +1716,10 @@ export CONFIG_RTL8189FS = m
 all: modules
 
 modules:
-	$(MAKE) ARCH=$(ARCH) CROSS_COMPILE=$(CROSS_COMPILE) -C $(KSRC) M=$(shell pwd)  modules
+	$(MAKE) ARCH=$(ARCH) CROSS_COMPILE=$(CROSS_COMPILE) -C $(KSRC) O=$(KBUILD_OUTPUT) M=$(PWD)  modules
+
+modules_install:
+	$(MAKE) ARCH=$(ARCH) CROSS_COMPILE=$(CROSS_COMPILE) -C $(KSRC) O=$(KBUILD_OUTPUT) M=$(PWD) INSTALL_MOD_PATH=$(INSTALL_MOD_PATH) modules_install
 
 strip:
 	$(CROSS_COMPILE)strip $(MODULE_NAME).ko --strip-unneeded
